docs/libvirt-pod-networking.md

```diff
@@ -4,7 +4,7 @@ Author: Fabian Deutsch \<fabiand@redhat.com\>
 
 ## Introduction
```

Comment:
The separation need can be traced back to existing host vs virt networking. One is concerned about node connectivity, the other about VM connectivity, each having their own scope and role, having different challenges to solve. The addition of the container level, is injected in between those, but it feels more closely related to the node connectivity, as it connects the infrastructure on which the VM/s are operating (but one can argue its a new 3rd level).
The current subneting that Kubernetes is exposing sees the pod as the endpoint, but now we suggest to extend it into the VM itself, which is moving the endpoint to the application itself that runs inside the container (which is beyond Kubernetes scope of management).
The services that can be exposed in the Kubernetes network are L3 based, which means you can let the VM access them from other networks as long as routing is available, I do not see a problem there.

If I understand how the IP pools are assigned, nodes have a subnet range, and each pod is assigned an address in that range. How will that play with a VM migrating to a different node? It cannot retain the same IP, it needs to switch to the other node subnet, but that will not allow a smooth migration.